### PR TITLE
VPN-6213 (part 5) - Change colors in code to non-specific names

### PR DESCRIPTION
**Pull request by mcleinman**
## Description

This is the fifth color-related PR. In this one:
- Color names used in code are changed to generic, non-color-specific names.
- Some color objects/names were removed as part of this process.
- The colors.js file is restructured - the sections will essentially become files in the next PR, and are not permanent.
- `MZToggleCard` and `MZCollapsableCard`, both unused, were removed. (This was done in consultation w/ @lesleyjanenorton.) 

Note: Some of the generic color names were judgement calls - I had to decide which generic name to use for each instance of a specific color in the code. While the colors look identical to `main`, I'm sure I didn't get this 100% perfect. 

For an oversimplified example, we used `.white` all over our codebase. In some cases that was replaced by `bgColorStronger`, in other places it's now a `card.defaultColor`. If a future theme doesn't also use the same color for both `card.defaultColor` and `bgColorStronger`, we may find a place where we need to swap `card.defaultColor` for a `bgColorStronger` in the code (or vice versa) - and in some cases possibly break one color name into two different ones. I'd expect tweaks to be done to this code when we eventually build out an additional theme.

---

This builds on https://github.com/mozilla-mobile/mozilla-vpn-client/pull/10019, which in turn builds on https://github.com/mozilla-mobile/mozilla-vpn-client/pull/10015, which in turn builds on https://github.com/mozilla-mobile/mozilla-vpn-client/pull/10014, which in turn builds on https://github.com/mozilla-mobile/mozilla-vpn-client/pull/9983. All 4 have been merged into `main`.

There will be one more color-related PR coming, built on top of this. It will better structure the files for swapping in themes.

## Reference

VPN-6213

## Checklist
    
- [x] My code follows the style guidelines for this project
- [x] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [x] I have performed a self review of my own code
- [x] I have commented my code PARTICULARLY in hard to understand areas
- [x] I have added thorough tests where needed

### CODE REVIEW

File: nebula/ui/components/MZSubscriptionOption.qml [hunk 1]
```diff
@@ -25,21 +25,21 @@ RadioDelegate {
...
         id: bg
-        color: MZTheme.colors.white
+        color: MZTheme.colors.bgColorStronger
         radius: MZTheme.theme.cornerRadius
 
         MZRectangularGlow {
             anchors.fill: bg
             glowRadius: radioDelegate.checked ? 8 : 1
             spread: radioDelegate.checked ? 0.1 : 0
-            color: MZTheme.colors.grey30
```
Comment (oskirby) [inline nebula/ui/components/MZSubscriptionOption.qml:35]: I think this is correct, but it's just different from most other drop shadows, which were `grey60`

**Review by strseb — APPROVED** [review]

File: nebula/ui/components/MZSubscriptionOption.qml (hunk 1 above)
Comment (mcleinman) [reply nebula/ui/components/MZSubscriptionOption.qml:35]: This was an inconsistency I streamlined when doing the PR. I tested it - it's a nearly imperceptible difference, and I took the liberty to slim down the colors used by reusing `grey60` here. Let me know if you don't like it, or want to see the side-by-side screenshots.

File: nebula/ui/components/MZBottomSheet.qml [hunk 2]
```diff
@@ -19,7 +19,7 @@ import Mozilla.Shared 1.0
               anchors.left: parent.left
               anchors.right: parent.right
               height: 20
-              color: MZTheme.colors.green90
```
[the remark this answers is not in the document]
Comment (mcleinman) [reply nebula/ui/components/MZBottomSheet.qml:22]: This is a comment, and is being used as an example. That said, the inconsistency is bugging me and we have `successMain` as the new `green90`, so I've updated this comment. :)

File: nebula/ui/components/MZAlert.qml [hunk 3]
```diff
@@ -53,10 +53,10 @@ Rectangle {
         id: style
         readonly property string darkCloseIcon: "qrc:/nebula/resources/close-dark.svg"
         readonly property string whiteCloseIcon: "qrc:/nebula/resources/close-white.svg"
-        property var alertColor: MZTheme.colors.black
-        property var alertHoverColor: MZTheme.colors.grey30
-        property var alertClickColor: MZTheme.colors.white
-        property var fontColor: MZTheme.colors.yellow10
+        property var alertColor: MZTheme.colors.normalLevelAccent
```
[the remark this answers is not in the document]
Comment (mcleinman) [reply nebula/ui/components/MZAlert.qml:56]: The change of values was intentional after. I checked with Basti, he confirmed these defaults are always overridden.

Good catch on the `alertClickColor` issue - I've updated this. Thank you very, very much for catching that. (Removing some unneeded colors was a fairly late change in this PR, and I fully missed that I dropped a color in use here.)

File: nebula/ui/components/MZGreyLink.qml [hunk 4]
```diff
@@ -9,5 +9,5 @@ import Mozilla.Shared 1.0
 MZLinkButton  {
     fontSize: MZTheme.theme.fontSizeSmallest
     Layout.preferredHeight: 18
-    linkColor: MZTheme.colors.greyLink
```
[the remark this answers is not in the document]
Comment (mcleinman) [reply nebula/ui/components/MZGreyLink.qml:12]: Oooh, great catch. I was grepping on color related things specifically. I've updated this to `MZTextLink`, as the color we're now using is also `textLink`.

**Comment by mcleinman** [comment]
@oskirby given that some changes were made (and those issues you found weren't all intentional), I'm requesting an r+ from you before I merge it.

**Review by oskirby — APPROVED** [review]
Looks good!

**Pull request merged**